pkg/bootkube/create.go

```diff
@@ -23,40 +23,34 @@ func CreateAssets(manifestDir string, timeout time.Duration) error {
 
 	upFn := func() (bool, error) {
 		if err := apiTest(); err != nil {
-			glog.Warningf("Unable to determine api-server version: %v", err)
+			glog.Warningf("Unable to determine api-server readiness: %v", err)
 			return false, nil
 		}
 		return true, nil
 	}
 
 	createFn := func() (bool, error) {
 		err := createAssets(manifestDir)
-		if err != nil {
-			glog.Errorf("Error creating assets: %v", err)
-			// If error is "system namespace not found" we should retry
-			if apierrors.IsNotFound(err) {
-				details := err.(*apierrors.StatusError).Status().Details
-				if details.Name == api.NamespaceSystem && details.Kind == "namespaces" {
-					return false, nil // retry
-				}
-			}
-			UserOutput("\nError creating assets: %v\n", err)
-			UserOutput("\nNOTE: Bootkube failed to create some cluster assets. It is important that manifest errors are resolved and resubmitted to the apiserver.\n")
-			UserOutput("For example, after resolving issues: kubectl create -f <failed-manifest>\n\n")
-		}
-		return true, nil
+		return true, err
```

Comment:
We weren't returning an error here before because there had been situations where an asset wasn't created - but it really wasn't a failure case, per-se (or you could actually recover it normally). An example of this was the dns service failed to be created because another service was assigned the service IP it was expecting to use.

The question was, should we let the cluster continue to boot - then let an admin resolve the issue (e.g. change the service IP of the conflicting service, and re-create the DNS service) -- VS -- just outright failing.

My inclination would be to keep the behavior the same for now, unless we have a good reason to change it - so either log the error here and return nil, or log in caller, but don't bubble it up further. No strong feelings for either.